mlos_bench/mlos_bench/optimizers/base_optimizer.py

```diff
@@ -91,7 +93,11 @@ def bulk_register(self, configs: Sequence[dict], scores: Sequence[Optional[float
             raise ValueError("Numbers of configs and scores do not match.")
         if status is not None and len(configs or []) != len(status or []):
             raise ValueError("Numbers of configs and status values do not match.")
-        return bool(configs and scores)
+        has_data = bool(configs and scores)
+        if has_data and self._start_with_defaults:
```

Comment:
Why not move that to `suggest()`?

Reply:
e.g. increase `self.iter` - the optimizer has been trained with that many samples already after all.

Reply:
Is the issue with interacting with how the base classes and abstract methods interact?

I feel like we should do this regardless though.
> e.g. increase self.iter - the optimizer has been trained with that many samples already after all.

Reply:
right now when we specify `max_iterations=N` in the config that means "train the optimizer for N more iterations", regardless of how many iterations we had ran before. I think it is a reasonable approach. It also makes sense when we implement merging of the experiments - then we just say, e.g., "merge 100 trials from this experiment and 50 trials from that experiment and run the optimizer for additional 50 iterations"

Reply:
Anyway, I would keep the existing logic for now - we can always change it later, and now we have lots of other more important things to do

Reply:
Kk, we should make it more clear that it's "N *more* iterations" somewhere. Maybe just update the description in the schema file for now.

Reply:
updated the description in the json schema